oauth2_provider/oauth2_backends.py

```diff
@@ -180,7 +180,11 @@ def extract_body(self, request):
         :return: provided POST parameters "urlencodable"
         """
         try:
-            body = json.loads(request.body.decode('utf-8')).items()
+            data = json.loads(request.body.decode('utf-8'))
+            if isinstance(data, dict):
+                body = data.items()
+            else:
+                body = ""
```

Comment:
This is the original code:

https://github.com/evonove/django-oauth-toolkit/blob/master/oauth2_provider/oauth2_backends.py#L182

It's setting empty string when it can't parse a JSON value. So empty string seems to be safe. When body is valid JSON but not a dictionary the old code would raise an exception (not intentionally) as it couldn't run `.items()` on a Python list. So if you use oauth-toolkit you can't send lists as body of your request even when it's absolutely irrelevant to oatuh-toolkit. The above code behaves with lists just similar to invalid JSONs as oauth-toolkit can't consume lists. But it shouldn't raise an exception. It should just ignore it.